### PR TITLE
[UTY-2390] CodeWriter v2: namespace, structs, classes, enums, methods, annotations (2/3)

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

- IAnnotatable
- NamespaceBlock
- TypeBlock (structs/classes)
- EnumBlock
- MethodBlock

#### Documentation

- [x] changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/EnumBlock.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public class EnumBlock : ScopeBlock, IAnnotatable
+    {
+        private static readonly string EnumMemberSeparator = $",{Environment.NewLine}";
+
+        internal EnumBlock(string declaration, Action<EnumBlock> populate) : base(declaration)
+        {
+            populate(this);
+        }
+
+        public void Member(string snippet)
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/EnumBlock.cs]: Fun fact, enum members are annotatable

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/EnumBlock.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        public void Members(ICollection<string> snippets)
+        {
+            Add(new TextList(EnumMemberSeparator, snippets));
+        }
+
+        public void Annotate(string annotation)
+        {
+            Annotation = annotation;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/EnumBlock.cs]: Is this `Annotation` on the base class?

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/TypeBlock.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        public void Line(ICollection<string> snippets)
+        {
+            if (snippets.Any())
+            {
+                Add(new TextList(snippets));
+            }
+        }
+
+        public void Initializer(string declaration, Func<IEnumerable<string>> populate)
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/TypeBlock.cs]: What does an initializer on a type block correspond to?

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/EnumBlock.cs (hunk 2 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/EnumBlock.cs]: ya

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/TypeBlock.cs (hunk 3 above)
Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/TypeBlock.cs]: discussed offline

**Pull request merged**